training/train_model.py

```diff
@@ -294,11 +300,17 @@ def train_model(self) -> None:
             if self._model.is_frozen() and self._epoch >= self._settings.training_settings.num_freeze_epochs:
                 logging.info("Unfreezing pre-trained model weights")
                 self._model.unfreeze_all()
+        if isinstance(self._writer, JsonWriter):
+            self._writer.flush()
 
     def evaluate_model(self, result_file: str, epoch: Optional[int] = None) -> None:
         if epoch is not None:
             self._load_model(file_name=os.path.join(self._settings.output_folder, _get_file_name(epoch)),
                              pretrained=True)
+        if self._settings.finetuned_epoch is not None:
+            self._load_model(file_name=os.path.join(self._settings.output_folder,
+                                                    _get_file_name(self._settings.finetuned_epoch)),
+                             pretrained=False)
         else:
```

Comment:
I want the PLP to be able to select model from epoch based on best validation performance. Common for models that are very flexible and can easily overfit such as deep neural networks and gradient boosted trees.